### PR TITLE
Navigation logic, reduce build times

### PR DESCRIPTION
**Pull request by gemfarmer**
Fixes issue(s) #2077

[CircleCI](https://circleci.com/gh/18F/18f.gsa.gov/tree/current-blog)

[:sunglasses: PREVIEW](https://federalist.18f.gov/preview/18F/18f.gsa.gov/current-blog/)

Changes proposed in this pull request:
- Fixes bug with authors, tags, and posts not indicating their status as 'current' navigation fields
- reduced build time significantly for the `./serve` command. It is down to ~10s
- I reduced build time by being pretty liberal about files and directories that would be `excluded` in the `config-dev.yml` file. The upside of this is that it is pretty easy to make things build again if necessary. The downside of this is that all things related to the blog are obscured from the build. This is because the slowest parts of the templating are the parts that filter from all the available posts. This will only get slower with time. A follow up issue could be to improve the filtering logic to create _slightly_ faster build times. I doubt that we would be able to reduce this _that much_ though, even if we spent a good chunk of time on it. 

<details open>
<summary>Page build times</summary>
<pre><code>
Filename                                                    | Count |   Bytes |  Time
------------------------------------------------------------+-------+---------+------
_layouts/default.html                                       |    56 | 674.31K | 1.139
_includes/navigation.html                                   |    56 | 283.42K | 0.720
_includes/header.html                                       |    56 |  61.53K | 0.329
_includes/navigation/drawer.html                            |    56 |  47.48K | 0.234
_includes/head.html                                         |    56 | 146.67K | 0.204
_includes/navigation/navbar.html                            |    56 |  26.48K | 0.125
_includes/navigation/footer-desktop.html                    |    56 |  98.50K | 0.083
_includes/navigation/footer-mobile.html                     |    56 | 108.01K | 0.078
_includes/javascript.html                                   |    56 |  52.50K | 0.032
_layouts/default-intro.html                                 |    42 |  66.54K | 0.030
pages/what-we-deliver.md                                    |     1 |   6.07K | 0.019
_layouts/default-image.html                                 |     9 |  22.44K | 0.016
feed.xml                                                    |     1 |   0.66K | 0.008
_layouts/project-tag-results.html                           |     1 |   2.57K | 0.005
_includes/feature-image.html                                |     9 |   0.37K | 0.004
_includes/newsletter.html                                   |    56 |  35.71K | 0.004
pages/humans.txt                                            |     1 |   0.04K | 0.003
blog/index.html                                             |     1 |   0.87K | 0.002
_includes/tag-results.html                                  |     1 |   0.16K | 0.002
_hire/partnership-playbook/index.md                         |     1 |   1.12K | 0.001
_layouts/tag-index.html                                     |     1 |   0.15K | 0.001
pages/april-fools/2016.md                                   |     1 |   0.55K | 0.001
pages/home.html                                             |     1 |   2.66K | 0.001
pages/april-fools/2015.md                                   |     1 |   0.50K | 0.000
pages/how-we-work.md                                        |     1 |   2.54K | 0.000
pages/press.html                                            |     1 |  13.11K | 0.000
pages/april-fools/index.md                                  |     1 |   0.25K | 0.000
feed/index.html                                             |     1 |   0.31K | 0.000
pages/500.md                                                |     1 |   0.13K | 0.000
_hire/partnership-playbook/we-build-in-the-open.md          |     1 |   4.09K | 0.000
pages/developer.md                                          |     1 |   3.80K | 0.000
_hire/partnership-playbook/we-build-in-the-open.md/#excerpt |     1 |   0.45K | 0.000
pages/developers.md                                         |     1 |   0.15K | 0.000
developers/index.html                                       |     1 |   0.32K | 0.000
pif/projects/round-2/index.html                             |     1 |   0.47K | 0.000
_projects/fec-gov.md                                        |     1 |   1.59K | 0.000
_hire/index.md                                              |     1 |   0.96K | 0.000
_hire/partnership-playbook/how-we-work.md/#excerpt          |     1 |   0.40K | 0.000
pages/pif/faq.html                                          |     1 |   0.00K | 0.000
pages/404.md                                                |     1 |   0.13K | 0.000
pages/thank-you.md                                          |     1 |   0.44K | 0.000
pages/pif/projects/index.md                                 |     1 |   2.06K | 0.000
_hire/partnership-playbook/who-we-work-with.md              |     1 |   2.79K | 0.000
_hire/partnership-playbook/how-we-work.md                   |     1 |   2.51K | 0.000
_projects/fec-gov.md/#excerpt                               |     1 |   0.13K | 0.000
_hire/index.md/#excerpt                                     |     1 |   0.11K | 0.000
pages/open-source-policy.md                                 |     1 |   7.86K | 0.000
tags/presidential innovation fellows/index.html             |     1 |   0.42K | 0.000
pages/dashboard.html                                        |     1 |   0.00K | 0.000
pages/feature-image-test.md                                 |     1 |   1.13K | 0.000

                    done in 8.192 seconds.
</code></pre>
</details>
/cc @coreycaitlin @mugizico

### CODE REVIEW

File: _includes/tag-results.html [hunk 1]
```diff
@@ -1,5 +1,4 @@
 {% assign posts = include.posts | default: site.posts %}
-{% assign tags = include.tags | default: post.tags | default: site.tags %}
```
Comment (gemfarmer) [inline _includes/tag-results.html]: Unused

File: _navigation.json [hunk 2]
```diff
@@ -1,30 +0,0 @@
----
```
Comment (gemfarmer) [inline _navigation.json]: Unused

**Comment by gemfarmer** [comment]
Ready for review!

**Comment by coreycaitlin** [comment]
Looks good, but after I run `./serve`, I get the following error in Terminal every time I navigate to another page locally: `ERROR `/assets/js/search-bundle.js' not found.`

**Comment by coreycaitlin** [comment]
^^ Just kidding, that's not a new issue! Carry on.

**Pull request merged**